xen-ocaml/core/build.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh -ex
+
+export PKG_CONFIG_PATH=`opam config var prefix`/lib/pkgconfig
+PKG_CONFIG_DEPS="openlibm libminios-xen >= 0.5"
+pkg-config --print-errors --exists ${PKG_CONFIG_DEPS} || exit 1
+
+case `uname -m` in
+armv*)
+  ARCH_CFLAGS="-DTARGET_arm"
+  m_header=m.arm.h
+ ;;
+*)
+  ARCH_CFLAGS="-DTARGET_amd64 -D__x86_64__ -momit-leaf-frame-pointer -mfancy-math-387"
+  m_header=m.x86_64.h
+  ;;
+esac
+
+# This extra flag only needed for gcc 4.8+
+# TODO we only need this for minios and not applications I think
+GCC_MVER2=`gcc -dumpversion | cut -f2 -d.`
+if [ $GCC_MVER2 -ge 8 ]; then
+  EXTRA_CFLAGS=-fno-tree-loop-distribute-patterns
+fi
```

Comment:
It doesn't appear to be exposed in the `pkg-config` CFLAGS though.  E.g. for `xen-ocaml/core` with the local test removed:

```
gcc -c -I../byterun -DCAML_NAME_SPACE -DNATIVE_CODE -DTARGET_amd64 -DSYS_linux  -pg -O -DPROFILING -Wno-attributes   -DSYS_xen -isystem/home/avsm/.opam/xendisk/lib/pkgconfig/../../include/minios-xen -isystem/home/avsm/.opam/xendisk/lib/pkgconfig/../../include/minios-xen/posix -m64 -mno-red-zone -fno-reorder-blocks -fno-asynchronous-unwind-tables -nostdinc -isystem /usr/lib/gcc/x86_64-linux-gnu/4.9/include -U__linux__ -U__FreeBSD__ -U__sun__ -U__linux -D__MINIOS__ -I/home/avsm/.opam/xendisk/include     -I /home/avsm/.opam/xendisk/include/mirage-xen/include    -DNATIVE_CODE -USYS_linux parsing.p.c
```

No `-fno-tree-loop-distribute-patterns`